postgen.js

```diff
@@ -21,9 +22,54 @@ export const postProcess = (bomNSData, options) => {
 
   bomNSData.bomJson = filterBom(jsonPayload, options);
   bomNSData.bomJson = applyStandards(bomNSData.bomJson, options);
+  bomNSData.bomJson = applyMetadata(bomNSData.bomJson, options);
   return bomNSData;
 };
 
+/**
+ * Apply additional metadata based on components
+ *
+ * @param {Object} bomJson BOM JSON Object
+ * @param {Object} options CLI options
+ *
+ * @returns {Object} Filtered BOM JSON
+ */
+export const applyMetadata = (bomJson, options) => {
+  const bomPkgTypes = new Set();
+  const bomPkgNamespaces = new Set();
+  for (const comp of bomJson.components) {
+    if (comp.purl) {
+      try {
+        const purlObj = PackageURL.fromString(comp.purl);
+        if (purlObj?.type) {
+          bomPkgTypes.add(purlObj.type);
+        }
+        if (purlObj?.namespace) {
+          bomPkgNamespaces.add(purlObj.namespace);
+        }
+      } catch (e) {
+        // ignore
+      }
+    }
+  }
+  if (!bomJson.metadata.properties) {
+    bomJson.metadata.properties = [];
+  }
+  if (bomPkgTypes.size) {
+    bomJson.metadata.properties.push({
```

Comment:
@cerrussell Added some extra properties to metadata, which is breaking the snapshots.

Reply:
@prabhu I have updated the properties. Running the snapshot tests locally passed, so I think you're probably good to go.

Reply:
Thank you! Once this is merged, it will be nice to have some tests with multiple types including excludes.